server/db/shop.test.js

```diff
@@ -0,0 +1,31 @@
+const knex = require('knex')
+const config = require('./knexfile').test
+const testDb = knex(config)
+
+const db = require('./shop')
+
+jest.setTimeout(10000)
+
+beforeAll(() => {
+  return testDb.migrate.latest()
+})
+
+beforeEach(() => {
+  return testDb.seed.run()
+})
+
+afterAll(() => {
+  return testDb.destroy()
+})
+
+describe('getProductsByGardens', () => {
+  it('gets all products by gardens id', () => {
```

Comment:
Awesome, great test.

As I alluded in my comment on the seeds, I think you could change this test to check the join has worked correctly, ie: it can find multiple products for a garden AND exclude products that aren't for the garden.

Think like a server-side engineer ;)  They love everything to be just-so.